ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -99,21 +100,30 @@ export class WorkflowLogsViewer extends React.Component<WorkflowLogsViewerProps,
         this.ensureUnsubscribed();
         this.subscription = services.workflows.getContainerLogs(this.props.workflow, this.props.nodeId, this.props.container, this.props.archived).subscribe(
             log => {
-                if (log) {
-                    this.setState(state => {
-                        log.split('\n').forEach(line => {
-                            state.lines.push(line);
-                        });
-                        return state;
-                    });
-                }
+                this.setState(state => {
+                    const newState = {...state, loaded: true};
+
+                    if (log) {
```

Comment:
Noting that I did not have this `if` statement in a previous version, and I had this fail with `undefined has no method split`.  That means the typings for this callback fn are wrong since it should likely be `string | undefined` rather than `string` so typescript can catch those types of errors.  Either that or the code emitting values to this stream should be emitting `""` rather than `undefined`.

Reply:
I did not know this.